client/my-sites/stats/stats-subscribers-overview/index.tsx

```diff
@@ -1,57 +1,113 @@
+
+const indexFirstCard = 0; // 0 for today
+const indexSecondCard = 30; // 30 days out
+const indexThirdCard = 60; // 60 days out
+const indexFourthCard = 90; // 90 days out
+
+interface SubscribersData {
+	period: string;
+	subscribers: number;
+	subscribers_change: number;
+}
+
+interface SubscribersDataResult {
+	data: SubscribersData[];
+	unit: string;
+	date: string;
+}
+
+interface SubscribersOverviewProps {
+	siteId: number;
+}
+
+// calculate the dates to query for
+const dateToday = calculateQueryDate( indexFirstCard );
+const date30DaysAgo = calculateQueryDate( indexSecondCard );
+const date60DaysAgo = calculateQueryDate( indexThirdCard );
+const date90DaysAgo = calculateQueryDate( indexFourthCard );
+
+// calculate the date to query for based on the number of days to subtract
+function calculateQueryDate( daysToSubtract: number ) {
+	const today = new Date();
+	const date = new Date( today );
+	date.setDate( date.getDate() - daysToSubtract );
+	return date.toISOString().split( 'T' )[ 0 ];
+}
+
+// calculate the stats to display in the cards
+function SubscribersOverviewCardStats( subscribersData: SubscribersData[][] ) {
+	const getCount = ( index: number ) => {
+		return subscribersData[ index ]?.[ 0 ]?.subscribers || 0;
+	};
 
-function SubscribersOverviewCardStats() {
 	const overviewCardStats = [
 		{
 			heading: translate( 'Today' ),
-			count: 1410,
+			count: getCount( 0 ),
 		},
 		{
 			heading: translate( '30 days ago' ),
-			count: 1080,
-			previousCount: 1080,
+			count: getCount( 1 ),
 		},
 		{
 			heading: translate( '60 days ago' ),
-			count: 1000,
-			previousCount: 1080,
+			count: getCount( 2 ),
 		},
 		{
 			heading: translate( '90 days ago' ),
-			count: 970,
-			previousCount: 1080,
+			count: getCount( 3 ),
 		},
 	];
+
 	return overviewCardStats;
 }
 
-function SubscribersOverviewCards() {
-	const overviewCardStats = SubscribersOverviewCardStats();
+const SubscribersOverview: React.FC< SubscribersOverviewProps > = ( { siteId } ) => {
+	const period = 'day';
+	const quantity = 1;
+	const dates: string[] = [ dateToday, date30DaysAgo, date60DaysAgo, date90DaysAgo ];
+	const subscribersQueries = useQueries( {
+		queries: dates.map( ( date ) => ( {
+			queryKey: [ 'stats', 'subscribers', siteId, period, quantity, date ],
+			queryFn: () => querySubscribers( siteId, period, quantity, date ),
+			select: selectSubscribers,
+			staleTime: 1000 * 60 * 5, // 5 minutes
+		} ) ),
+	} ) as UseQueryResult< SubscribersDataResult >[];
 
-	return (
-		<div className="overview-cards-list">
-			{ overviewCardStats.map( ( overviewCardStat ) => (
-				<CountComparisonCard
-					key={ overviewCardStat.heading }
-					heading={ overviewCardStat.heading }
-					count={ overviewCardStat.count }
-					previousCount={ overviewCardStat.previousCount }
-					showValueTooltip
-					icon={ false }
-				/>
-			) ) }
-		</div>
-	);
-}
+	const isLoading = subscribersQueries.some( ( result ) => result.isLoading );
```

Comment:
What do you say if we deal the errors and loading status separately, because we still want to show the working cards even when some of them errored out or are still loading.